### PR TITLE
LPD-20765 Add test case for Object Proxy Salesforce boolean field

### PR DESCRIPTION
**Pull request by jairmedeiros**
Related issue: https://liferay.atlassian.net/browse/LPD-20765

### CODE REVIEW

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -461,7 +486,7 @@ protected Page<ObjectEntry> getObjectEntries(
 	}
 
 	private ObjectEntry _addObjectEntry(
-			String customStatus, Date date, String title)
+			String customStatus, Date date, String title, boolean flagged)
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:489]: Hi @jairmedeiros, `flagged` parameter should come before `title`, please sort this.


Thanks!

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -353,6 +370,14 @@ public void testGetObjectEntries() throws Exception {
 
 		// Equals/not equals expression
 
+		testGetObjectEntries(
+			HashMapBuilder.put(
+				"filter",
+				filterString.concat(
+					buildEqualsExpressionFilterString("flagged", true))
+			).build(),
+			objectEntry2, objectEntry4);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:379]: Hey @jairmedeiros this use case should come after [this one](https://github.com/liferay-objects/liferay-portal/pull/3174/files#diff-82f88640eff05dedc724f92e76be2cf21cd6e4ffae22f9cba25e62e361ced5d0R406-R412), sort this as well,

Thank you!

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -214,6 +215,20 @@ public void setUp() throws Exception {
...
+				"Flagged__c"
+			).userId(
+				adminUser.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap("Flagged")
+			).name(
+				"flagged"
+			).objectDefinitionId(
+				_objectDefinition.getObjectDefinitionId()
+			).build());
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:230]: Please sort this as well, it should come after [this](https://github.com/liferay-objects/liferay-portal/pull/3174/files#diff-82f88640eff05dedc724f92e76be2cf21cd6e4ffae22f9cba25e62e361ced5d0R160-R172)

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @jairmedeiros see my comments,

Nice job BTW! :tada:

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

nice job @jairmedeiros

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
Hey @MarinhoFeliphe @guilhermedcamacho, for [this](https://github.com/liferay-objects/liferay-portal/pull/3174#issuecomment-2018334200) relevant failure, you should add a Flagged__c field (Checkbox type -- it's like boolean) on Salesforce side.

**Comment by MarinhoFeliphe** [comment]
`Flagged__c` custom field added on SF side

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/149035

**Pull request closed without merging**